src/rdkafka.h

```diff
@@ -7889,6 +7979,349 @@ rd_kafka_DeleteRecords_result_offsets(
 
 /**@}*/
 
+/**
+ * @name Admin API - DescribeTopics
+ * @{
+ */
+
+/**
+ * @brief DescribeTopics result type.
+ *
+ */
+typedef struct rd_kafka_TopicDescription_s rd_kafka_TopicDescription_t;
+
+/**
+ * @brief Describe topics as specified by the \p topics
+ *        array of size \p topics_cnt elements.
+ *
+ * @param rk Client instance.
+ * @param topics Array of topics to describe.
+ * @param topics_cnt Number of elements in \p topics array.
+ * @param options Optional admin options, or NULL for defaults.
+ *                Valid options:
+ *                 - include_authorized_operations
+ * @param rkqu Queue to emit result on.
+ *
+ * @remark The result event type emitted on the supplied queue is of type
+ *         \c RD_KAFKA_EVENT_DESCRIBETOPICS_RESULT
+ */
+RD_EXPORT
+void rd_kafka_DescribeTopics(rd_kafka_t *rk,
+                             const char **topics,
```

Comment:
Can receive a `rd_kafka_TopicCollection_t` that currently is created only from topic names. In future with topic ids too. See J.

Reply:
Currently it contains only one field as char **topics and init method rd_kafka_TopicCollection_new_from_names,  later we can make it into a union.